heron/instance/src/java/com/twitter/heron/instance/bolt/BoltInstance.java

```diff
@@ -57,6 +58,8 @@
   // The bolt will read Data tuples from streamInQueue
   private final Communicator<Message> streamInQueue;
 
+  private State<byte[], byte[]> instanceState;
```

Comment:
Why are State types being limited to byte[] here? A bolt implementor should be able to define state as anything that satisfies `State<? extends Serializable, ? extends Serializable>` no?